### PR TITLE
feat: add decorator for CanBeBanned

### PR DESCRIPTION
**Pull request by sirasitxp**
- [X] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [X] My pull request targets the `main` branch of Chapter.

Closes part of #1217(canBeBanned)

### CODE REVIEW

File: server/prisma/generator/factories/instanceRoles.factory.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import { prisma } from '../../../src/prisma';
 
-import { InstancePermission, Permission } from '../../../../common/permissions';
+import {
+  ChapterPermission,
+  InstancePermission,
+  Permission,
+} from '../../../../common/permissions';
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/instanceRoles.factory.ts:7]:
```suggestion
import { InstancePermission, Permission } from '../../../../common/permissions';
```
We members should not be able to ban users, so we do not need to pull out the chapter permissions here.

File: server/prisma/generator/factories/instanceRoles.factory.ts [hunk 2]
```diff
@@ -20,6 +24,7 @@ const roles: InstanceRole[] = [
     permissions: [
       InstancePermission.ChapterJoin,
       InstancePermission.ChapterSubscriptionsManage,
+      ChapterPermission.ChapterBanUser,
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/instanceRoles.factory.ts:27]:
```suggestion
```
As mentioned, this isn't something members should be able to do.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hi, @hisnameispum, thanks for working on this. Could you restore the chapter page? It seems to have been deleted by accident.

Also, to avoid regressions it would be great to have tests that confirm that the permissions work. Namely that users who should not be able to ban actually cannot.

https://github.com/freeCodeCamp/chapter/pull/1346 is a good example of this.

**Comment by ojeytonwilliams** [comment]
Hi @hisnameispum if you've got a moment, could you take a look at the failing tests?

**Comment by Sboonny** [comment]
@ojeytonwilliams, I am wondering how to find admin role in users because I can't find it.

image


I think the cypress test about finding admin role fails, because it doesn't exist in user dashboard

**Comment by ojeytonwilliams** [comment]
> I am wondering how to find admin role in users because I can't find it.

There aren't instance admins (yet), since we figured they weren't MVP worthy.  The roles are instance owner, instance member, chapter admin and chapter member.

> after fixing status error, banned admin don't have permission to view the table at all

Yep, that's right. I'll update the tests.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**